intel_extension_for_transformers/llm/runtime/graph/models/llama/llama.cpp

```diff
@@ -181,7 +182,7 @@ static bool llama_model_eval_internal(model_context& lctx, const model_token* to
     Qcur = ne_rope_inplace(ctx0, Qcur, std::max(n_cached - N, n_past), n_rot, 0, 0);
     ne_set_name(Qcur, "Qcur");
     Kcur = ne_rope_inplace(  // n_ctx exceeds but it will be shift-roped back with cached K
-        ctx0, Kcur, (n_past == n_total ? n_past : n_ctx), n_rot, 0, 0);
+        ctx0, Kcur, (is_full_ring ? n_ctx : n_past), n_rot, 0, 0);
```

Comment:
following the above, just need to determine whether n_past or n_total
```suggestion
        ctx0, Kcur, (shift_roped_k && n_total > n_past ? n_ctx : n_past), n_rot, 0, 0);
```